tests/test_20_use_case_topology.py

```diff
@@ -229,5 +229,114 @@ def test_011_intra_domain_link_down_path_found(self):
+                    "vlan": "300",
+                },
+                {
+                    "port_id": "urn:sdx:port:tenet.ac.za:Tenet01:50",
+                    "vlan": "300",
+                }
+            ]
+        }
+        response = requests.post(l2vpn_api_url, json=l2vpn_payload)
+        assert response.status_code == 201, response.text
+        l2vpn_id = response.json().get("service_id")
+
+        # Wait for L2VPN to be provisioned
+        time.sleep(5)
+
+        response = requests.get(l2vpn_api_url)
+        assert response.status_code == 200, response.text
+        l2vpn_status = response.json().get(l2vpn_id).get("status")
+        assert l2vpn_status == "up", f"L2VPN status should be up, but is {l2vpn_status}"
+
+        h1, h6 = self.net.net.get('h1', 'h6')
+        h1.cmd('ip link add link %s name vlan300 type vlan id 300' % (h1.intfNames()[0]))
+        h1.cmd('ip link set up vlan300')
+        h1.cmd('ip addr add 10.3.1.1/24 dev vlan300')
+        h6.cmd('ip link add link %s name vlan300 type vlan id 300' % (h6.intfNames()[0]))
+        h6.cmd('ip link set up vlan300')
+        h6.cmd('ip addr add 10.3.1.6/24 dev vlan300')
+
+        # test connectivity
+        assert ', 0% packet loss,' in h1.cmd('ping -c4 10.3.1.6')
+
+        node = self.net.net.get('Ampath1')
+        config = node.cmd('ovs-vsctl get-controller', node.name).split()
+        set_node(node, 'down', "tcp:127.0.0.1:6654")
+
+        time.sleep(15)
+        
+        # status of the node should be down
+        api_url_topo = SDX_CONTROLLER + '/topology'
+        response = requests.get(api_url_topo)
+        assert response.status_code == 200, response.text
+        nodes = response.json().get('nodes', [])
+        for n in nodes:
+            if n['name'] == node:
+                assert n['status'] == "down", f"Node {n['name']} status should be down, but is {n['status']}"
+                break
+
+        response = requests.get(l2vpn_api_url)
+        l2vpn_status = response.json().get(l2vpn_id).get("status")
+        assert l2vpn_status == "down", f"L2VPN status should be error, but is {l2vpn_status}"
+
+        # test connectivity
+        assert ', 100% packet loss,' in h1.cmd('ping -c4 10.3.1.6')
+
+        # Step 5: Attempt to provision a new L2VPN using the down node
+        new_l2vpn_payload = {
+            "name": "Test L2VPN for node down - should fail",
+            "endpoints": [
+                {
+                    "port_id": "urn:sdx:port:ampath.net:Ampath1:50",
+                    "vlan": "301",
+                },
+                {
+                    "port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50",
+                    "vlan": "301",
+                }
+            ]
+        }
+        response = requests.post(l2vpn_api_url, json=new_l2vpn_payload)
+        assert response.status_code != 201, "Should not be able to provision L2VPN on a down node"
+
+        ### Reset
+        set_node(node, 'up', " ".join(config))
```

Comment:
This reset being executed after the assertions can lead to a failures and the node remain down for other future tests. I advise you to set the status to UP again before any assertion. (right after the sleep you can get the data from topology, set to UP again and only then execute the asserts)